src/utils.ts

```diff
@@ -21,6 +21,34 @@ export const mac = ios || /Mac/.test(platform);
 * Utility functions used in one or more files
 */
 
+/**
+ * @internal
+ * @returns a number, representing ReturnType<typeof requestAnimationFrame>
+ * Expose a scheduler for after react's render cycle is over. Some
+ * internal functions use it, and testing infrastructure may use it as well
+ * see stackoverflow.com/questions/26556436/react-after-render-code/28748160#28748160
+ * 
+ * - If the owner argument is passed, an inner timeout of 0 is set as 'pendingTimeout'
+ *   so that it can be cancelled by the owner later
+ * - If an extraDelay is passed, the inner timeout waits Xms after the render cycle
+ */
+export type afterDOMUpdateHandle = {
+  raf: number,
+  timeout: ReturnType<typeof setTimeout>
+};
+export function setAfterDOMUpdate(f: ()=>void, extraDelay?:number): afterDOMUpdateHandle {
+  let timeoutHandle: ReturnType<typeof setTimeout>;
+  return {
+    raf: window.requestAnimationFrame(() => timeoutHandle = setTimeout(f, extraDelay)),
+    timeout: timeoutHandle
+  };
+}
+
+export function cancelAfterDOMUpdate(handle:afterDOMUpdateHandle): void {
+  cancelAnimationFrame(handle.raf);
+  clearTimeout(handle.timeout);
+}
```

Comment:
This will throw a runtime error when a null or undefined handle is passed in, and none of the code in `componentWillUnmount` is checking to see if the handle is defined. To match `clearTimeout` behavior, you should check if handle exists and just do nothing if it doesn't.

Reply:
Won't the default value (`{}`) solve this problem? Or is that just really inelegant?

Reply:
Looks like I replied to an outdated commit that was missing the default value. Yes the default value takes care of this.